BrainlyDemoDay/BrainlyDemoDay/SubscriptionManager/SubscriptionManager.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+
+public enum SubscriptionState {
+    case monthly(Date)
+    case limitedEntries(Int)
+    case empty
+}
+
+public protocol SubscriptionManagerProtocol {
+    func fetchSubscriptionFor(tutoringUser user: TutoringSDKUser, completion: @escaping (Swift.Result<SubscriptionState, Error>) -> Void)
+}
+
+public protocol Networking {
+
+}
+
+public final class SubscriptionManager: SubscriptionManagerProtocol {
+
+    private let networkManager: Networking
+
+    init(networkManager: Networking) {
+        self.networkManager = networkManager
+    }
+
+    public func fetchSubscriptionFor(tutoringUser user: TutoringSDKUser, completion: @escaping (Swift.Result<SubscriptionState, Error>) -> Void) {
```

Comment:
Passing `TutoringSDKUser` requires this module to be tightly coupled with `TutoringSDK`, which I think is not expected. I would use a proxy object to identify a "subscription user".